warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -2057,54 +2058,63 @@ protected Future<IteratorSetting> loadQueryIterator(final MetadataHelper metadat
                         configureAdditionalOptions(config, cfg);
                         
                         try {
-                            addOption(cfg, QueryOptions.INDEX_ONLY_FIELDS,
-                                            QueryOptions.buildFieldStringFromSet(metadataHelper.getIndexOnlyFields(config.getDatatypeFilter())), true);
-                            addOption(cfg, QueryOptions.COMPOSITE_FIELDS,
-                                            QueryOptions.buildFieldStringFromSet(metadataHelper.getCompositeToFieldMap(config.getDatatypeFilter()).keySet()),
-                                            true);
-                            addOption(cfg, QueryOptions.INDEXED_FIELDS,
-                                            QueryOptions.buildFieldStringFromSet(metadataHelper.getIndexedFields(config.getDatatypeFilter())), true);
-                        } catch (TableNotFoundException e) {
-                            QueryException qe = new QueryException(DatawaveErrorCode.INDEX_ONLY_FIELDS_RETRIEVAL_ERROR, e);
-                            throw new DatawaveQueryException(qe);
+                            Set<String> compositeFields = metadataHelper.getCompositeToFieldMap(config.getDatatypeFilter()).keySet();
+                            Set<String> indexedFields = metadataHelper.getIndexedFields(config.getDatatypeFilter());
+                            Set<String> indexOnlyFields = metadataHelper.getIndexOnlyFields(config.getDatatypeFilter());
+                            
+                            // only reduce the query fields if planning has occurred
+                        if (!isPreload && config.getReduceQueryFields()) {
+                            Set<String> queryFields = ReduceFields.getQueryFields(config.getQueryTree());
```

Comment:
I am very confused.  It makes perfect sense to reduce the fields in the VisitorFunction, but why reduce them here?

Reply:
Performing an initial intersection here reduces the number of elements intersected in the visitor function.